### ISSUE TITLE
Logs do not flush on rotation

### ISSUE THREAD

**Issue by realpdm** [apache/trafficserver#2544]
When ATS rotates log files, log entries will continue to be written to the rotated log file name for a period of time. For a low volume log file we observed for up to 30 minutes the old file received log entries.  This amounted to about 130 log lines and 85043 bytes of data.

For a high volume log this is barely noticeable and only 2-3 seconds of log lines may be written to the rotated file.

It seems like maybe ATS is not flushing its log buffers and opening new file after rotation so the old file is able to receive  events for a period of time.

We are running ATS 7.1.0.

**Comment by phonehold** [comment]
@realpdm  try this patch, It works for me
```
diff --git a/lib/ts/BaseLogFile.cc b/lib/ts/BaseLogFile.cc
index bd4d2d6..338e96d 100644
--- a/lib/ts/BaseLogFile.cc
+++ b/lib/ts/BaseLogFile.cc
@@ -138,6 +138,8 @@ BaseLogFile::roll(long interval_start, long interval_end)
     m_meta_info = new BaseMetaInfo(m_name.get());
   }
 
+  close_file();
+
   // Create the new file name, which consists of a timestamp and rolled
   // extension added to the previous file name.  The timestamp format is
   // ".%Y%m%d.%Hh%Mm%Ss-%Y%m%d.%Hh%Mm%Ss", where the two date/time values
```

**Comment by oknet** [comment]
please check the TS-4072 and commit 346b4192e4004c0cbfb93b34965a0394212c35b4.

**Comment by pbchou** [comment]
Hi all, I created a PR for this issue last week. Appreciate it if some one participating in this issue can take a look -- #4354 -- please see comments in the commit for my analysis. Thanks. This is a critical issue for us as our log flow depends upon 30-second rolled files, and this was broken between 6.2.1 and 7.1.4.

**Comment by bryancall** [comment]
It looks like this was resolved in 8.0.3
